tasks/java/config/agent.go

```diff
@@ -185,17 +186,57 @@ func checkConfig(configs []config.ConfigElement) ([]config.ConfigElement, bool)
 
 // This check looks for the existence of the newrelic.jar in the file system as a final attempt at identifying this as a java app present
 func checkForJar() bool {
-	jarNames := []string{
-		"newrelic.jar",
+	//check for existence of this specific file name in the user's system. We do not look for a regex/pattern name because it would be a never ending search
+	if tasks.FileExists("newrelic.jar") {
+		log.Debug("Jar file found, setting true")
+		return true
+	}
+	//Now we can attempt to find a newrelic filename pattern, but only in the current directory
+	jarRgx := regexp.MustCompile(`(newrelic)?i([\S]+)?\.jar`)
+	dirRgx := regexp.MustCompile(`(newrelic)?i`) //in case they put the jar inside a newrelic directory
```

Comment:
I'm not certain the ?i portion of the regex is doing what we want here.
It seems to be allowing for `'newrelic' or 'any string with an i in it'`
Playing around with it here: https://golang.org/pkg/regexp/

It looks like adding parenthesis around (?i) if the idea is that the `.jar` is meant to be case insensitive
And moving (?i) to the front of `newrelic` if we want that to be case insenstitive.

Code snippet that I've been testing on: 
```
package main

import (
	"fmt"
	"regexp"
)

func main() {
	// Compile the expression once, usually at init time.
	// Use raw strings to avoid having to quote the backslashes.
	var validID = regexp.MustCompile(`(i?)(newrelic)([\S]+)?\.jar`)

	fmt.Println(validID.MatchString("bluerelxc.jar"))
	fmt.Println(validID.MatchString("bluerelic.jar"))
	fmt.Println(validID.MatchString("newrelic.jar"))
	fmt.Println(validID.MatchString("newRelic.jar"))
	fmt.Println(validID.MatchString("nerwRelic.JAR"))
}
```